Taskfile.yml

```
@@ -11,10 +11,84 @@ includes:
     optional: true
     taskfile: ./bindings/go/runtime/Taskfile.yml
     dir: ./bindings/go/runtime
+  bindings/go/generator:
+    optional: true
+    taskfile: ./bindings/go/generator/Taskfile.yml
+    dir: ./bindings/go/generator
 
+vars:
+  GO_MODULES:
+    sh: find {{ .ROOT_DIR }}/bindings/go -name go.mod -exec dirname {} \; | sed 's|{{ .ROOT_DIR }}/bindings/go/||'
 
 tasks:
   test:
+    desc: "Run all tests in the project"
+    cmds:
+      - for: { var: GO_MODULES }
+        task: bindings/go/{{.ITEM}}:test
+
+  init/go.work:
+    desc: "Initialize the workspace environment"
+    status:
+      - find go.work
+    cmds:
+      - go work init
+      - for: { var: GO_MODULES }
+        cmd: 'go work use bindings/go/{{.ITEM}}'
+      - go work sync
+
+  generate:
+    desc: "Run all Code Generators in the project"
+    cmds:
+      - task: 'bindings/go/generator:ocmtypegen/generate'
+      - task: 'controller-gen/generate-deepcopy'
+
+  ensure-clean:
+    desc: "Ensure the project is clean by checking for uncommitted changes"
+    cmds:
+      - git diff --exit-code
+      - git status --porcelain
+
+  controller-gen/generate-deepcopy:
+    desc: "Generate kubernetes-style deepcopy code for all types in the project"
+    deps: [controller-gen/install]
+    cmds:
+      - 'tmp/bin/controller-gen object paths=./bindings/go/generator/ocmtypegen/test/...'
+
+  controller-gen/install:
+    desc: "Install {{.PKG}} at {{ .TARGET_VERSION }} into tmp directory if not already present"
+    vars:
+      PKG: sigs.k8s.io/controller-tools/cmd/controller-gen
+      TARGET_VERSION: v0.17.2
+      CURRENT_VERSION:
+        sh: 'tmp/bin/controller-gen --version || echo "0.0.0"'
+    status:
+    - '{{ contains .TARGET_VERSION .CURRENT_VERSION }}'
+    env:
+      GOBIN: '{{ .ROOT_DIR }}/tmp/bin'
+    cmds:
+      - go install {{ .PKG }}@{{ .TARGET_VERSION }}
+
+  golangci-lint/run:
+    desc: "Run golangci-lint on all go modules"
+    deps: [golangci-lint/install]
+    cmds:
+      - for: { var: GO_MODULES }
+        cmd: |
+          cd {{ .ROOT_DIR }}/bindings/go/{{.ITEM}} && 
+          {{ .ROOT_DIR }}/tmp/bin/golangci-lint run \
+            --timeout 10m \
+            --config={{ .ROOT_DIR }}/.github/config/golangci.yml \
+            --path-prefix {{ .ROOT_DIR }}/bindings/go/{{.ITEM}} \
+            {{ .ROOT_DIR }}/bindings/go/{{.ITEM}}/...
+
+  golangci-lint/install:
+    desc: "Install golangci-lint at {{ .TARGET_VERSION }} into tmp directory if not already present"
+    vars:
+      TARGET_VERSION: 1.64.7
+      CURRENT_VERSION:
+        sh: 'tmp/bin/golangci-lint version --format short || echo "0.0.0"'
+    status:
+      - '{{ contains .TARGET_VERSION .CURRENT_VERSION }}'
     cmds:
-      - task: bindings/go/descriptor:test
-      - task: bindings/go/runtime:test
+      - curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/HEAD/install.sh | sh -s -- -b {{ .ROOT_DIR }}/tmp/bin v{{ .TARGET_VERSION }}
```

The link points to the latest version of the script and is directly piped into a shell. If the script is updated with malicious code we wouldn't notice. 

Usually this should not be a problem, but it increases the attack surface

---

I would like to take this as a calculated risk since its for local testing and CI on isolated runners only. Acceptable to you?

---

Maybe we can meet in between

```suggestion
      - curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/refs/tags/v{{ .TARGET_VERSION }}/install.sh | sh -s -- -b {{ .ROOT_DIR }}/tmp/bin v{{ .TARGET_VERSION }}
```

This would at least decrease the attack surface a bit, as the version is pinned (and by this, also an official release).

---

Works for me 👍 :)